arangod/Cluster/ShardDistributionReporter.cpp

```diff
@@ -424,10 +418,59 @@ void ShardDistributionReporter::getDistributionForDatabase(
         }
       }
 
-      ReportOffSync(col.get(), allShards.get(), counters, aliases, result);
+      ReportOffSync(col.get(), allShards.get(), counters, aliases, result, progress);
       todoSyncStateCheck.pop();
     }
   }
+}
+
+void ShardDistributionReporter::getCollectionDistributionForDatabase(
+    std::string const& dbName, std::string const& colName, VPackBuilder& result) {
+
+  double endtime = TRI_microtime() + 2.0; // We add two seconds
+
+  result.openObject();
+
+  auto aliases = _ci->getServerAliases();
+  auto col = _ci->getCollection(dbName, colName);
+
+  std::queue<std::shared_ptr<LogicalCollection>> todoSyncStateCheck;
+
+  auto allShards = col->shardIds();
+  if (testAllShardsInSync(dbName, col.get(), allShards.get())) {
+    ReportInSync(col.get(), allShards.get(), aliases, result);
+  } else {
+    todoSyncStateCheck.push(col);
+  }
+
+  const bool progress = true;
+  helperDistributionForDatabase(dbName, result, todoSyncStateCheck, endtime, aliases, progress);
+  result.close();
+}
+
+void ShardDistributionReporter::getDistributionForDatabase(
+    std::string const& dbName, VPackBuilder& result) {
+
+  double endtime = TRI_microtime() + 2.0; // We add two seconds
+
+  result.openObject();
+
+  auto aliases = _ci->getServerAliases();
+  auto cols = _ci->getCollections(dbName);
+
+  std::queue<std::shared_ptr<LogicalCollection>> todoSyncStateCheck;
+
+  for (auto col : cols) {
+    auto allShards = col->shardIds();
+    if (testAllShardsInSync(dbName, col.get(), allShards.get())) {
+      ReportInSync(col.get(), allShards.get(), aliases, result);
+    } else {
+      todoSyncStateCheck.push(col);
+    }
+  }
+
+  const bool progress = false;
+  helperDistributionForDatabase(dbName, result, todoSyncStateCheck, endtime, aliases, progress);
   result.close();
 }
```

Comment:
Is it possible to combine this two functions?
The forLoop + queue mgmt is identical.
Maybe we can get sth that takes the queue + the LogicalCollection as input.
@hkernbach Easier to explain in Person if you have questions

Reply:
ok, lets discuss tomorrom. I'm not sure:
1. uses col as string
2. uses cols as vector etc.

Reply:
no need to change it here.
approved by @mchacki